### ISSUE TITLE
indentation rule inserts unnecessary spacings inside multiline if-else condition

### ISSUE THREAD

**Issue by romtsn** [pinterest/ktlint#871]
## Expected Behavior
Expect this code to remain unchanged after reformatting
```Kotlin
fun function(): Boolean {
    return if (
        listOf(
            param1,
            param2,
            param3
        ).none { it != null }
    ) {
        true
    } else {
        false
    }
}
```

## Observed Behavior
```Kotlin
fun function(): Boolean {
    return if (
        listOf(
                param1,
                param2,
                param3
            ).none { it != null }
    ) {
        true
    } else {
        false
    }
}
```

## Steps to Reproduce
Try to format the given example, indentation rule will insert unnecessary spacings before arguments and rbrace

## Your Environment

* Version of ktlint used: 0.38.0
* Name and version (or code for custom task) of integration used (Gradle plugin, Maven plugin, command line, custom Gradle task): CLI

**Comment by fishy** [comment]
I'm seeing a similar issue on `0.38.1`.

I have the code of:

```kotlin
        if (
            pref.getBoolean(
                SettingsActivity.KEY_BIO_AUTOLOAD,
                SettingsActivity.DEFAULT_BIO_AUTOLOAD
            )
        ) {
```

`ktlint` would complain about `Unexpected indentation (16) (should be 20)` on the two args lines and closing parenthesis line. `ktlint -F` would try to reformat it as:

```kotlin
        if (
            pref.getBoolean(
                    SettingsActivity.KEY_BIO_AUTOLOAD,
                    SettingsActivity.DEFAULT_BIO_AUTOLOAD
                )
        ) {
```

And still complain about those 3 lines.

**Comment by aljidy** [comment]
I get the same issue here on `0.38.1`  with this:

```
  onNextAction?.let {
                setOnEditorActionListener { _, actionId, _ ->
                    if (actionId in arrayOf(
                            IME_ACTION_NEXT,
                            IME_ACTION_DONE
                        )
                    ) {
                        it.invoke()
                        true
                    } else {
                        false
                    }
                }
            }
```

it alternates between these two sets of warnings depending on the indents:
```
file.kt:85:29: Unexpected indentation (expected 24, actual 28)
file.kt:86:29: Unexpected indentation (expected 24, actual 28)
file.kt:87:25: Unexpected indentation (expected 20, actual 24)
```

```
file.kt:85:1: Unexpected indentation (24) (should be 28)
file.kt:86:1: Unexpected indentation (24) (should be 28)
file.ktt:87:1: Unexpected indentation (20) (should be 24)
```

**Comment by grrowit** [comment]
I can confirm it is happening still with latest version
